depthai_sdk/src/depthai_sdk/managers/encoding_manager.py

```diff
@@ -48,16 +50,24 @@ def createDefaultQueues(self, device):
 
         for cameraName, node in self._encodingNodes.items():
             self._encodingQueues[cameraName] = device.getOutputQueue(cameraName + "EncXout", maxSize=30, blocking=True)
-            suffix = ".h265" if node.getProfile() == dai.VideoEncoderProperties.Profile.H265_MAIN else ".h264"
-            self._encodingFiles[cameraName] = (self.encodeOutput / cameraName).with_suffix(suffix).open('wb')
+            # suffix = ".h265" if node.getProfile() == dai.VideoEncoderProperties.Profile.H265_MAIN else ".h264"
+            if self.encodeConfig[cameraName][1] == ".mcap":
+                self._mcap[cameraName] = DepthAiMcap(cameraName)
```

Comment:
DepthAiMcap should call `imageInit` during initialization as it requires the same parameters. After adding `close` method to it, you'll be able to use `_encodingFiles` to store an object of this class